### PR TITLE
Fixes #26975 - Errata/package repository filter

### PR DESCRIPTION
**Pull request by sjha4**
**To Test**

1. Create a yum repository "Repo1" with errata content. (ex: https://fedorapeople.org/groups/katello/fakerepos/zoo)
2. Create a yum repository "Repo2" without errata content. (ex: https://repos.fedorapeople.org/repos/pulp/pulp/fixtures/rpm-with-vendor/)
3. Sync Repo1 and Repo2
4. Go to Content > Errata > 
5. In the repository filter, you should see only Repo1.
Earlier, you would see both Repo1 and Repo2.

You can also create an empty yum repo and verify that it doesn't show up in either Errata or Packages page repository filter.

### CODE REVIEW

**Comment by jturel** [comment]
I like this a lot. I worked with ATIX folks on something similar on another open PR: https://github.com/Katello/katello/pull/7961 I'll ask that they update to use this approach once merged

File: app/models/katello/repository.rb [hunk 1]
```diff
@@ -123,6 +123,7 @@ class Repository < Katello::Model
     scope :archived, -> { where('environment_id is NULL') }
     scope :in_published_environments, -> { in_content_views(Katello::ContentView.non_default).where.not(:environment_id => nil) }
     scope :order_by_root, ->(attr) { joins(:root).order("#{Katello::RootRepository.table_name}.#{attr}") }
+    scope :with_content, ->(content) { where(:id => Katello::RepositoryTypeManager.find_content_type(content).model_class.repository_association_class.select(:repository_id)) }
```
Comment (jturel) [inline app/models/katello/repository.rb]: I think it could be possible to do a join here which is probably better than a subquery. What do you think?

Comment (sjha4) [reply app/models/katello/repository.rb]: @jturel : How does this look to you:
joins(Katello::RepositoryTypeManager.find_content_type(content).model_class.repository_association_class.name.demodulize.underscore.pluralize.to_sym)

File: app/controllers/katello/api/v2/repositories_controller.rb [hunk 2]
```diff
@@ -90,6 +90,7 @@ def custom_index_relation(collection)
     param :description, String, :desc => N_("description of the repository")
     param :available_for, String, :desc => N_("interpret specified object to return only Repositories that can be associated with specified object.  Only 'content_view' & 'content_view_version' are supported."),
           :required => false
+    param :with_content, String, :desc => N_("only repositories having at least one of the specified content type ex: rpm , erratum")
```
Comment (jturel) [inline app/controllers/katello/api/v2/repositories_controller.rb]: Do we have any precedent in the code that can autogenerate the available types for the api docs? I seem to recall something but not sure where

File: app/models/katello/repository.rb [hunk 3]
```diff
@@ -123,6 +123,7 @@ class Repository < Katello::Model
     scope :archived, -> { where('environment_id is NULL') }
     scope :in_published_environments, -> { in_content_views(Katello::ContentView.non_default).where.not(:environment_id => nil) }
     scope :order_by_root, ->(attr) { joins(:root).order("#{Katello::RootRepository.table_name}.#{attr}") }
+    scope :with_content, ->(content) { joins(Katello::RepositoryTypeManager.find_content_type(content).model_class.repository_association_class.name.demodulize.underscore.pluralize.to_sym) }
```
Comment (jturel) [inline app/models/katello/repository.rb]: This looks good - but it needs a .distinct :+1:

File: app/controllers/katello/api/v2/repositories_controller.rb (hunk 2 above)
Comment (sjha4) [reply app/controllers/katello/api/v2/repositories_controller.rb]: Right..I thought about that but couldn't see a straightforward way to get all content_types..I'll think about this a little..

File: app/models/katello/repository.rb (hunk 3 above)
Comment (sjha4) [reply app/models/katello/repository.rb]: Added..

File: app/controllers/katello/api/v2/repositories_controller.rb [hunk 4]
```diff
@@ -90,6 +90,7 @@ def custom_index_relation(collection)
     param :description, String, :desc => N_("description of the repository")
     param :available_for, String, :desc => N_("interpret specified object to return only Repositories that can be associated with specified object.  Only 'content_view' & 'content_view_version' are supported."),
           :required => false
+    param :with_content, RepositoryTypeManager.enabled_content_types, :desc => N_("only repositories having at least one of the specified content type ex: rpm , erratum")
     param_group :search, Api::V2::ApiController
```
Comment (sjha4) [inline app/controllers/katello/api/v2/repositories_controller.rb]: Added allowed values here. API looks like: 
with_content

**Pull request merged**